### PR TITLE
Update vendored k8s to 1.9

### LINKED ISSUE #200: rest_client_requests_total metric missing
The rest_client_requests_total is missing:
- it is included in client-go
- but the prometheus metric itself is dependency injected from `k8s.io/kubernetes/pkg/client/metrics/prometheus`, which is an old version on head and refers to `k8s.io/kubernetes/pkg/client/metrics` and not `k8s.io/client-go/tools/metrics`

### PR DESCRIPTION
**Pull request by tomwilkie**
Update the vendored kubernetes to 1.9, the client-go code to a matching version etc.

Also switch from godep to dep.  Godep is **dep**recated.

Fixes #200

### CODE REVIEW

**Comment by tomwilkie** [comment]
Turns out this doesn't work, we really need this patch from 1.9

**Comment by tomwilkie** [comment]
It works!

screen shot 2018-02-28 at 14 58 32

**Comment by MrHohn** [comment]
Fixes #213.

File: README.md [hunk 1]
```diff
@@ -36,7 +36,6 @@ This is the repository for [Kubernetes DNS](http://kubernetes.io/docs/admin/dns/
 ## Release process
 
 1. Build and test (`make build` and `make test`)
-1. Update [go dependencies](docs/go-dependencies.md) if needed.
```
Comment (MrHohn) [inline README.md]: Hoping dep could be good enough so that we won't need a substitute of the godep script we have...

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  name = "github.com/spf13/pflag"
+  version = "1.0.0"
+
+[[constraint]]
+  name = "github.com/stretchr/testify"
+  version = "1.2.1"
+
+[[constraint]]
+  branch = "master"
+  name = "k8s.io/apimachinery"
```
Comment (MrHohn) [inline Gopkg.toml]: Make it also version = "kubernetes-1.9.3" (matching the other constraints)?

File: Gopkg.toml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+[[constraint]]
+  branch = "master"
+  name = "k8s.io/apiserver"
```
Comment (MrHohn) [inline Gopkg.toml]: Make it also version = "kubernetes-1.9.3" (matching the other constraints)?

File: Gopkg.toml [hunk 4]
```diff
@@ -0,0 +1,82 @@
+# Gopkg.toml example
```
Comment (MrHohn) [inline Gopkg.toml]: Remove the example template.

File: Gopkg.toml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  branch = "master"
+  name = "github.com/skynetservices/skydns"
+
+[[constraint]]
+  name = "github.com/spf13/pflag"
+  version = "1.0.0"
+
+[[constraint]]
+  name = "github.com/stretchr/testify"
+  version = "1.2.1"
```
Comment (MrHohn) [inline Gopkg.toml]: I don't quite get above constraints (expect `github.com/skynetservices/skydns` as we do need the latest commit - required by https://github.com/kubernetes/dns/pull/219, and `github.com/prometheus/client_golang` - guessing required by https://github.com/kubernetes/dns/issues/200). Could you explain? Thanks.

**Pull request closed without merging**